DOCS-2300 Add TOTP Documentation (#11277)
DataDog/documentation: content/en/synthetics/guide/app-that-requires-login.md

title: Running tests on an application that requires authentication
kind: guide
further_reading:
  - link: '/synthetics/guide/browser-tests-totp'
    tag: 'Documentation'

>     text: 'Time-based one time passwords (TOTPs) for multi-factor authentication (MFA)'

Reviewer: ```suggestion
    text: 'Time-based one time passwords (TOTPs) for multi-factor authentication (MFA) in Browser tests'
```

Some MFA providers may detect Datadog's browser tests as bots and prevent them from logging in, for instance, by adding a reCAPTCHA. In this case, contact your MFA provider to see if it is possible to turn off bot detection when [identifying requests as coming from Synthetic browser tests][3] (such as for a specific set of credentials or Synthetic tests specific headers).

> If your MFA process involves steps performed outside of the browser, such as voice, text message, or opening a mobile application that does not leverage TOTP, also consider reaching out to your MFA provider to ask if your MFA settings could be modified or if MFA could be turned off when [identifying requests as coming from Synthetic browser tests][3] (for example, for a specific set of credentials, Synthetic tests specific headers, etc.) for testing purposes.

Reviewer 2: ```suggestion
If your MFA process involves steps performed outside of the browser (such as voice, text message, or opening a mobile application that does not leverage TOTP), consider reaching out to your MFA provider to ask if your MFA settings can be modified or if MFA can be turned off when [identifying requests as coming from Synthetic browser tests][3] (such as for a specific set of credentials or Synthetic tests specific headers) for testing purposes.
```

> Depending on the type of MFA leveraged by your application, [JavaScript steps][4] could help to work around that.

Reviewer 2: ```suggestion
Depending on the type of MFA leveraged by your application, [JavaScript steps][4] can help to work around that.
```
